account/payments.go

```diff
@@ -340,69 +618,106 @@ func (a *Service) getMaxAmount(destination string, routeHints []*lnrpc.RouteHint
 		a.log.Errorf("lnclient.ListChannels error: %v", err)
 		return 0, fmt.Errorf("lnclient.ListChannels error: %w", err)
 	}
+
 	var totalMax uint64
+OUTER:
 	for _, c := range channels.Channels {
-		if c.LocalBalance == 0 {
+		// Calculate how much can be sent through this channel after accounting for channel reserve
+		thisChannelCanSendSat, minHtlcMsat := channelConstraints(c)
```

Comment:
It seems that we are iterating over channels and querying routes to the destination calculating fees in a similar way we are doing in sendPaymentSelfSplit.
Do you see a way to extract out the code that calculate the routes and reuse it in both functions?
Perhaps getMaxAmount will return the amount and the routes to be used by sendPayment?

Reply:
I'll see what I can share. I started with shared code, but ended up breaking it up for ease of mind while developing. They're slightly different, because the `getMaxAmount` gets the maximum sendable amount, while the `sendPaymentSelfSplit` function sends whatever amount the caller requests.

Reply:
Updated this to share more code. It's not entirely overlapping, but the bulk is taken from subfunctions now.